### ISSUE TITLE
about the activity prediction

### ISSUE THREAD

**Issue by cpystan** [google/next-prediction#23]
In your paper, you use mAP to evaluate your activity prediction. 
To compute the mAP, you first compute AP of 30 classes.
But i want to try negative log likelihood. It is different.
I need to get the probability distribution of every sample.
I notive that in your code batch.data['future_activity_oneshot'](64 multiply 30)
and future_act (64 multiply 30) are related to the activity prediction?
So how can i get the probability distribution of every sample/batch about 30 classes? 
THANK YOU.

**Comment by JunweiLiang** [comment]
The output of activity prediction is unscaled logit of shape [batch_size, num_class] and it is trained with sigmoid_cross_entropy (multi-label). You can add a softmax activation to get a probability distribution.

**Comment by cpystan** [comment]
Add a softmax to future_act or batch.data['future_activity_oneshot']?
thank you very much.

**Comment by JunweiLiang** [comment]
Add softmax on the prediction outputs [here](https://github.com/google/next-prediction/blob/master/code/models.py#L452)

**Comment by cpystan** [comment]
I am trying to compute it in test_data, so I add softmax here https://github.com/google/next-prediction/blob/906024cfae939f32b928f3e7412a0bf266692ff1/code/test.py#L149 , but it is only (64 multiply 30). But The total batch number is 535*64. So how can i get the prediction output every epoch?
THANK YOU

**Comment by JunweiLiang** [comment]
I don't understand your reference and your question. Why would you add softmax to the test_data, which is the input of the model? To get the prediction output, you can refer to [this repo](https://github.com/JunweiLiang/social-distancing-prediction/blob/master/code/inference/main.py#L1037).
